### DISCUSSION TITLE
App ID not updating in apple app site association

### DISCUSSION THREAD

**Discussion by Travisswop** [vercel/community#2037] [Help]
### Summary

I updated the appID in apple-app-site-association which is in the .well-known folder. But when I access the file it still giving old appID. Why it is not updating in the server?

**Comment by amyegan** [comment]
Hi @Travisswop! Can you walk me through the steps you took to update the appID? New values should show up after a new deployment or redeployment. If that isn't working, I need to reproduce the problem to get a better look at what is going wrong. If you're able to share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that would be very helpful for debugging this problem 🙏

**Reply by Travisswop** [reply]
Hi @amyegan, In my nextjs project, I had a .well-known folder at the root of the project. In the apple-app-site-association file, inside the .well-known folder, I updated the appId. But when I browse, https://mysite.com/.well-known/apple-app-site-association, it downloads the JSON file which contains the old appID. 

Here is an example JSON - 
```
{
    "applinks": {
        "apps": [],
        "details": [
            {
                "appID": "updated app ID",
                "paths": [ "*" ]
            }
        ]
    }
}
```

**Reply by amyegan** [reply]
How recently did you make this change? And do you have any custom caching rules set up for your Next app?

I wasn't able to replicate the problem in my test app. Please share a minimal reproduction, sample code, and/or screenshots to help me get a better look at the problem

**Reply by Travisswop** [reply]
No, I don't have any custom caching rules. Just change the "appID" in the apple-app-site-association JSON file and redeploy the project.

Here is the screenshot of my project folder and JSON file - 
![Screenshot 2023-04-13 at 9 52 07 AM](https://user-images.githubusercontent.com/98964561/231646127-6f6c48c3-0567-4650-833a-56caa4ba3494.png)

You can download the JSON file at this link - https://swopme.app/.well-known/apple-app-site-association

You will see appID is not matching in the given screenshot and the server file.

**Reply by amyegan** [reply]
This may have the same cause as another [recent discussion](https://github.com/orgs/vercel/discussions/1808#discussioncomment-5604656) topic. Can you try adding a custom cache handler?

- Cache handler: https://github.com/vercel/next.js/blob/afd7a50a7780a56c2ebe1e3dea77d79054bf72f1/test/e2e/app-dir/app-static/cache-handler.js 
- Next config: https://github.com/vercel/next.js/blob/afd7a50a7780a56c2ebe1e3dea77d79054bf72f1/test/e2e/app-dir/app-static/next.config.js#L4 (don't forget to set `CUSTOM_CACHE_HANDLER: true`)

If it works, that will tell us that this is being caused by the same problem

**Reply by Travisswop** [reply]
I followed your suggestion but still, it gives old appID!
